packages/contractkit/src/utils/signing-utils.ts

```diff
@@ -213,3 +222,41 @@ export function decodeSig(sig: any) {
     s: ethUtil.toBuffer(s) as Buffer,
   }
 }
+
+/**
+ * Attempts each recovery key to find a match
+ */
+export function recoverKeyIndex(
+  signature: Uint8Array,
+  publicKey: BigNumber,
+  hash: Uint8Array
+): number {
+  for (let i = 0; i < 4; i++) {
+    const compressed = false
+    // Force types to be Uint8Array
+    const signatureArray = new Uint8Array(signature)
+    const hashArray = new Uint8Array(hash)
+    const recoveredPublicKeyByteArr = ecdsaRecover(signatureArray, i, hashArray, compressed)
+    const publicKeyBuff = Buffer.from(recoveredPublicKeyByteArr)
+    const recoveredPublicKey = bufferToBigNumber(publicKeyBuff)
+    console.log(
```

Comment:
not sure if `console.log` is used in contractkit, I think instead `debug` from line 14 is used

Reply:
good catch